tests/federation_room_join_partial_state_test.go

```diff
@@ -188,6 +188,249 @@ func TestPartialStateJoin(t *testing.T) {
 		t.Logf("Alice sent event event ID %s", eventID)
 	})
 
+	// we should be able to receive typing EDU over federation during the resync
+	t.Run("CanReceiveTypingDuringPartialStateJoin", func(t *testing.T) {
+		deployment := Deploy(t, b.BlueprintAlice)
+		defer deployment.Destroy(t)
+		alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+		server := createTestServer(t, deployment)
+		cancel := server.Listen()
+		defer cancel()
+		serverRoom := createTestRoom(t, server, alice.GetDefaultRoomVersion(t))
+		psjResult := beginPartialStateJoin(t, server, serverRoom, alice)
+		defer psjResult.Destroy()
+
+		derekUserId := psjResult.Server.UserID("derek")
+
+		content, _ := json.Marshal(map[string]interface{}{
+			"room_id": serverRoom.RoomID,
+			"user_id": derekUserId,
+			"typing":  true,
+		})
+		edu := gomatrixserverlib.EDU{
+			Type:    "m.typing",
+			Content: content,
+		}
+		psjResult.Server.MustSendTransaction(t, deployment, "hs1", []json.RawMessage{}, []gomatrixserverlib.EDU{edu})
+
+		psjResult.FinishStateRequest()
```

Comment:
Given https://github.com/matrix-org/synapse/issues/13684, do we want to test that we _don't_ block on incoming EDUs?

eg.
```go
		event := psjResult.CreateMessageEvent(t, "charlie", nil)
		serverRoom.AddEvent(event)
		server.MustSendTransaction(t, deployment, "hs1", []json.RawMessage{event.JSON()}, nil)
		awaitEventViaSync(t, alice, serverRoom.RoomID, event.EventID(), "")
```
ought to get stuck and fail the test if we block after the incoming EDU `/send`.

Reply:
Good idea indeed, I'll add that, thanks !

Reply:
Done in 7331ebc